sklearn/metrics/_classification.py

```diff
@@ -141,6 +147,27 @@ def _weighted_sum(sample_score, sample_weight, normalize=False):
         return sample_score.sum()
 
 
+def _nan_average(scores: np.ndarray, weights: Optional[np.ndarray]):
+    """
+    Wrapper for np.average, with np.nan values being ignored from the average
+    This is similar to np.nanmean, but allowing to pass weights as in np.average
+    """
+
+    mask = np.isnan(scores)
+    if mask.all():
+        return np.nan
+    if weights is None:
+        return np.average(scores[~mask])
+    if isinstance(weights, list):
+        weights = np.array(weights)
+
+    scores, weights = scores[~mask], weights[~mask]
+
+    if (weights == 0).all():
+        return np.average(scores)
```

Comment:
Checking for `weights == 0` adds more computation for an edge case. Can we pass weights directly into `np.average` and not do this check?

Reply:
unfortunately
```
ZeroDivisionError
When all weights along axis are zero. 
```
see https://numpy.org/doc/stable/reference/generated/numpy.average.html. But will change into try/except